homeassistant/components/bunq/sensor.py

```diff
@@ -0,0 +1,146 @@
+"""Support for bunq account balance."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from bunq.sdk.context import ApiContext, ApiEnvironmentType, BunqContext
+from bunq.sdk.exception import ApiException, BunqException
+from bunq.sdk.model.generated import endpoint
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+ICON = "mdi:cash-multiple"
+UPDATE_INTERVAL = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_API_KEY): cv.string})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up bunq sensors."""
+    sensors = []
+
+    try:
+        # setup api context
+        api_context = ApiContext(
+            ApiEnvironmentType.PRODUCTION,
+            config.config[CONF_API_KEY],
+            "Home Assistant",
+        )
+        api_context.ensure_session_active()
+        BunqContext.load_api_context(api_context)
+    except (ApiException, BunqException) as err:
+        _LOGGER.error(
+            "Error setting up sensor: %s", err,
+        )
+        raise PlatformNotReady
+
+    # create sensors
+    for account in get_account_data():
+        sensors.append(BunqBalanceSensor(account))
+
+    async_add_entities(sensors, True)
+
+    # schedule updates for sensors
+    data = BunqData(hass, api_context, sensors)
+    await data.schedule_update(UPDATE_INTERVAL)
+
+
+def get_account_data():
+    """Get active bunq accounts."""
+    active_accounts = []
+
+    accounts = endpoint.MonetaryAccountBank.list().value
```

Comment:
See https://developers.home-assistant.io/docs/en/integration_fetching_data.html for our recommended way to fetch and update data.

This looks like there's a single api that provides all entity data, so we should use the `DataUpdateCoordinator`.
https://developers.home-assistant.io/docs/en/integration_fetching_data.html#coordinated-single-api-poll-for-data-for-all-entities